File: EDA/semana_1/Pruebas/python src/selectionSort.py
from timer import Timer

def selection_sort(array):  
    for i in range(len(array)-1):
        primer = i
        for j in range(i+1, len(array)):
            if array[primer] > array[j]:
                primer = j

        array[primer], array[i] = array[i], array[primer]

if __name__ == "__main__":
       
    index = 0
    numArrays = int(input())
    for i in range(numArrays):
        promedio = 0
        arr = []    
        n = int(input())
        for j in range(2):
            aux = input().split(" ")
            for k in range(n):
                arr.append(int(aux[k]))

            t=Timer(n)   
            t.start()
            selection_sort(arr)
            t.stop()
            promedio += t.printTime()
        print(n,round(promedio/10,3))
        
       

File: EDA/semana_1/Pruebas/c++ src/selectionSort.cpp
#include <iostream>
#include <chrono>
#include <ctime>
#include "Timer.hpp"

using namespace std;

void selection_sort(int* array, int n){
    int primer;
    int temp;
    for(int i = 0; i < n-1; i++)
    {
        primer = i;
        for (int j = i+1; j < n; j++)
        {
            if(array[primer] > array[j])
                primer = j;
        }
        temp = array[primer];
        array[primer] = array[i];
        array[i] = temp;
    }
}

void print(int* array, int n){
    for(int i=0;i<n;i++){
        cout<<array[i]<<" ";
    }
}

int main()
{
    int numArrays;
    cin>>numArrays;

    for (int i = 0; i < numArrays; ++i)
    {
        int n;
        cin>>n;
        int* arr = new int [n];
        for (int i = 0; i < n; ++i)
            cin>>arr[i];
        Timer timer;
        timer.start();
        selection_sort(arr,n);
        timer.stop();
        cout<<n<<' '<<timer.elapsedMilliseconds()<<endl;
        delete [] arr;
    }
    
    return 0;
}
